src/cpp/include/nodegui/core/Events/eventwidget_macro.h

```diff
@@ -16,30 +16,45 @@
   COMPONENT_WRAPPED_METHODS_DECLARATION                              \
   Napi::Value initNodeEventEmitter(const Napi::CallbackInfo& info) { \
     Napi::Env env = info.Env();                                      \
-    this->instance->emitOnNode =                                     \
-        Napi::Persistent(info[0].As<Napi::Function>());              \
-    this->instance->connectSignalsToEventEmitter();                  \
+    EventWidget* eventWidget =                                       \
+        dynamic_cast<EventWidget*>(this->instance.data());           \
+    if (eventWidget) {                                               \
+      eventWidget->emitOnNode =                                      \
+          Napi::Persistent(info[0].As<Napi::Function>());            \
+      eventWidget->connectSignalsToEventEmitter();                   \
```

Comment:
`connectSignalsToEventEmitter()` needs to be virtual otherwise this line will call `EventWidget::connectSignalsToEventEmitter()` for every widget type and the events will break.